### PR TITLE
Solution by izifresh

### PR DESCRIPTION
**Pull request by izifresh**

### CODE REVIEW

**Comment by manwar** [comment]
Thanks @izifresh for sharing the solutions.

Would you like to receive the weekly challenge by email?
If yes then please send one line email to perlweeklychallenge@yahoo.com.

File: challenge-030/izifresh/Perl 5/christmas [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env perl
```
Comment (k-mx) [inline challenge-030/izifresh/Perl 5/christmas]: 1. Script must show _only_ Christmas that will be at sunday. Not all future Christmases
2. Try to wrap "magic numbers" into constants (`perldoc constant`), and your code will become more readable.
Or you can try to use hash as "day number" <-> "day name" mapping
3. change dir name and add extensions:
challenge-030/izifresh/Perl 5/christmas 
challenge-030/izifresh/perl5/christmas.pl
4. Stick with single indenting style.

File: challenge-030/izifresh/Perl 5/christmas [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    $year++;
+     if ($i == 4) {
+      $i = 1;
+      $x += 2;
+}
+     else { 
+      $x += 1;
+      $i++;
+}
+  if ($x == 7) { $x = 0; }
```
Comment (k-mx) [inline challenge-030/izifresh/Perl 5/christmas]: Just a suggestion:
You don't need trailing semicolon for such statements:
`if ($x == 7) { $x = 0 }`
Moreover, you can use short postfix form:
`$x = 0 if $x == 7;`

File: challenge-030/izifresh/Perl 5/christmas [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env perl
+package Christmas;
```
Comment (k-mx) [inline challenge-030/izifresh/Perl 5/christmas]: package is useless here

File: challenge-030/izifresh/Perl 5/numbers [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+package Numbers; 
+
+use strict; 
+use warnings;
+use v5.10;
+ for my $i (0..10) {
+   for my $j (0..10) {
+     for my $k (0..10) { 
+       if (($i+$j+$k) == 12) {
+         if ( $i % 2 == 0 or $j % 2 == 0 or $k % 2 == 0 ) {
```
Comment (k-mx) [inline challenge-030/izifresh/Perl 5/numbers]: DRY (Don't repeat yourself)!
`if ( $i % 2 == 0 or $j % 2 == 0 or $k % 2 == 0 ) {`
Shorter and more expressive version:
`if ( grep { $_ % 2 == 0 } $i, $j, $k ) {`
But the coolest source code is the source code that you can read as usual text, so, we can use vanilla module to become even more expressive:
```
# perldoc List::Util
use List::Util qw(any);
...
if ( any { $_ % 2 == 0 } $i, $j, $k ) {
```

File: challenge-030/izifresh/Perl 5/numbers [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+           my @m = ($i,$j,$k);
+           @m = sort { $a <=> $b } @m;    
+           print ("@m\n");
```
Comment (k-mx) [inline challenge-030/izifresh/Perl 5/numbers]: Omit brackets, they are useless here.

File: challenge-030/izifresh/Perl 5/christmas (hunk 1 above)
Comment (k-mx) [reply challenge-030/izifresh/Perl 5/christmas]: My bad @izifresh. Solutions must be named ch-1.pl and ch-2.pl

**Pull request merged**